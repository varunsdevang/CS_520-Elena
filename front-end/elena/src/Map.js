import { GoogleMap, Marker, useLoadScript } from "@react-google-maps/api";
import { useMemo } from "react";
import "./App.css";

const Map = () => {
  const { isLoaded } = useLoadScript({
    googleMapsApiKey: 'paste-ur-key-here',
  });
  const center = useMemo(() => ({ lat: 42.3732, lng: -72.519 }), []);

  return (
    <div className="Map">
      {!isLoaded ? (
        <h1>Loading...</h1>
      ) : (
        <GoogleMap
          center={center}
          zoom={10}
          mapContainerClassName="map-container"
        >
          <Marker position={{ lat: 18.52043, lng: 73.856743 }} />
        </GoogleMap>
      )}
    </div>
);
};

export default Map;