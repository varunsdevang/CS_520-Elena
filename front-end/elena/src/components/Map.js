import { GoogleMap, Marker, useLoadScript, Polyline } from "@react-google-maps/api";
import { useEffect, useState } from "react";
import "../App.css";

const Map = (props) => {
  const {route} = props;
  const { isLoaded } = useLoadScript({
    // googleMapsApiKey: 'AIzaSyB7szZ54ue7G5mZX-R0yDKo6aw2vvxzL60', //Varun API
    // googleMapsApiKey: 'AIzaSyCGP0U5PmCV0ZYtrlvOGQ3sdaWMi05LQt4'  //Prathiksha API
  });

  const [state, setState] = useState({
    center: { lat: 42.3732, lng: -72.519 },
    currentLocation: { lat: 42.3732, lng: -72.519},
    zoom: 10,
  })

  useEffect(() => {
    navigator.geolocation.getCurrentPosition((position)=>{

      setState({...state, currentLocation: {lat: position.coords.latitude, lng: position.coords.longitude}})
      console.log('position', position);
    })
  }, [])

  useEffect(() => {
    //route = route.map((coords) => new google.maps.LatLng(coords.lat, coords.lng))
    console.log('route', route);
    setState({...state, zoom: 13, center: route[1]})
  }, [route])

  return (
    <div className="Map">
      {!isLoaded ? (
        <h1>Loading...</h1>
      ) : (
        <GoogleMap
          center={state.center}
          zoom={state.zoom}
          mapContainerClassName="map-container"
        >
          <Marker position={{ lat: state.currentLocation.lat, lng: state.currentLocation.lng }} />
         <Polyline path={route} visible={true}  strokeColor={'#006aff'} strokeOpacity={1.0} strokeWeight={2}></Polyline>
        </GoogleMap>
      )}
    </div>
);
};

export default Map;