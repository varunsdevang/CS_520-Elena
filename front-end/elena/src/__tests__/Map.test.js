import { render, screen } from '@testing-library/react';
import React from "react";
import renderer from "react-test-renderer";
import Map from '../Components/Map';


describe("Map snapshot should match", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create(< Map/>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
});