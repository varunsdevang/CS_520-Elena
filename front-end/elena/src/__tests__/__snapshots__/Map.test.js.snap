// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map snapshot should match Matches DOM Snapshot 1`] = `
<div
  className="Map"
>
  <h1>
    Loading...
  </h1>
</div>
`;
